### PR TITLE
US137580 - Quizzing - Save score per question

### PR DESCRIPTION
**Pull request by maddiegabriel**
[US137580 - [Consistent Eval][Quizzing][UI] Hook up quiz questions to save question score](https://rally1.rallydev.com/#/42960320374d/custom/236803223728?detail=%2Fuserstory%2F630713522257)

### CODE REVIEW

File: components/consistent-evaluation-page.js [hunk 1]
```diff
@@ -1211,6 +1212,38 @@ export default class ConsistentEvaluationPage extends SkeletonMixin(LocalizeCons
 		);
 	}
 
+	async _transientSaveQuestionScore(e) {
```
Comment (maddiegabriel) [inline components/consistent-evaluation-page.js:1215]: doing pretty much the same thing as `_transientSaveQuestionFeedback()` here

File: components/consistent-evaluation-page.js [hunk 2]
```diff
@@ -1211,6 +1212,38 @@ export default class ConsistentEvaluationPage extends SkeletonMixin(LocalizeCons
...
+		await this._mutex.dispatch(
+			async() => {
+				let questionResponseEntity;
+				const newScore = e.detail.score;
+
+				const evaluationEntity = await this._controller.fetchEvaluationEntity(false);
```
Comment (jeroach) [inline components/consistent-evaluation-page.js:1221]: Is this block of code the same? If so can we move it to a helper method to reduce code duplication and aid maintainability?

File: components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js [hunk 3]
```diff
@@ -124,12 +124,26 @@ export class ConsistentEvaluationQuizzingAttemptResult extends SkeletonMixin(Loc
...
 			detail: {
-				feedback: feedback,
+				feedback: newFeedback,
+				questionResponseHref: this.questionResponseHref
+			}
+		}));
+	}
+
+	_emitGradeChangeEvent(newScore) {
+		if (newScore < 0) {
```
Comment (jeroach) [inline components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js:139]: Can score be set to 0 and still be a new score? 

i.e. should this be `newScore <= 0` ?

**Review by jeroach — APPROVED** [review]
lgtm

Just one question around a possible helper method.

File: components/consistent-evaluation-page.js (hunk 2 above)
Comment (maddiegabriel) [reply components/consistent-evaluation-page.js:1221]: good idea, i'll split some of that out 👍

**Review by JamieTran — APPROVED** [review]

**Pull request merged**